modules/statistics/php/stats_demographic.class.inc

```diff
@@ -374,100 +387,104 @@ class Stats_Demographic extends \NDB_Form
         }
 
         //START BIG TABLE
-        $inst_dropdown   = array_merge(
+        $instrumentList = \Utility::getAllInstruments();
+        $instDropdown   = array_merge(
             array( '' => 'Recruit Breakdown by Sex'),
-            \Utility::getAllInstruments()
+            $instrumentList
         );
-        $instrument_list = \Utility::getAllInstruments();
-        $this->tpl_data['all_instruments'] = $instrument_list;
-        if ($_REQUEST['DemographicInstrument'] ?? ''
-        ) {
-            $Subcategories = array(
-                              'Complete',
-                              'Incomplete',
-                             );
-            $result        = $DB->pselect(
+        $this->tpl_data['all_instruments'] = $instrumentList;
+        $demographicInstrument = $_REQUEST['DemographicInstrument'] ?? '';
+        if (!empty($demographicInstrument)) {
+            $demographicInstrument = htmlspecialchars($demographicInstrument);
```

Comment:
@ZainVirani can you explain why are you are adding the !empty and htmlspecialchars? i'm trying to understand why the "big table" is breaking

Reply:
`!empty($demographicInstrument)` is the equivalent to `$_REQUEST['DemographicInstrument'] ?? ''`

This PR just rewrites code, it isn't supposed to change functionality. If I did something it's because the previous code did the same thing but in a messy way.